app/src/main/java/org/gdg/frisbee/android/appwidget/UpcomingEventWidgetProvider.java

```diff
@@ -123,7 +123,18 @@ public void onNotFound(String key) {
                 });
 
         }
-
+        private Event getNextEvent(List<Event> listEvents) {
+            if (listEvents == null || listEvents.size() == 0) {
+                return null;
+            }
+            Event ret = listEvents.get(0);
```

Comment:
I didn't mean that we should separate them in the backend. Unfortunately, backend is really hard to touch. 

At some point last year, we separated the past and future events in the chapter events page. I don't know the exact specification of the backend but it returns past events as well. If you go to the main page in the app, you will see the past events. 

We basically sorted the list to seprate the past and future events. The issue described in #628 was a side effect of that separation. 

This method is not clear and feels like a patch for that side effect. Don't get me wrong, your code is fine. It's just that the codebase is not in perfect situation and we need help. 

And lastly, in this method, I still don't see the current date in the comparison. How do you make sure that the event returned here is the closest future event?